chore: adding a migration page (#222)
mercator-ocean/copernicus-marine-toolbox: doc/migration-guide/v2.rst

.. warning::
    One of the important breaking changes is that there is no prompt to confirm the download. Please use ``--dry-run`` if you do not want to download the files.
    The ``--force-download`` option has been removed.

General
------------------

Option ``--force-download`` deleted
""""""""""""""""""""""""""""""""""""""""

Before, it was necessary to accept a prompt to download the data.
Such prompt could be skipped adding the option ``--force-download`` at the end of the call.
In v2, the toolbox will download all the data requested without a prompt.

If you do not want to download the data but rather check what would be downloaded and if the query is correct,
you can use the option ``--dry-run`` and a response object will be returned with
the information of the data that would be downloaded.

.. code-block:: bash

    # In v1
    > copernicusmarine get --dataset-id cmems_mod_ibi_phy_my_0.083deg-3D_P1Y-m
    ...
    Do you want to download the data? [y/N]:

    # In v2
    > copernicusmarine get --dataset-id cmems_mod_ibi_phy_my_0.083deg-3D_P1Y-m
    ...

>     {

Author: do you think that maybe here doing a subset is better? (we would like people to use it instead of the get, no? (better handling of the data and faster and quicker to download)

      "number_of_files_to_download": 29,
      "total_size": 241.22591304779053,
      "status": "000",
      "message": "The request was successful."
    }

.. warning::

>   Without the ``--dry-run`` option, the data will be downloaded even if the query is very consequent.

Author: not sure of the use of the word 'consequent' here.

                {
                  "services": [
                    {
                      "service_name": "original-files",
                      "uri": "https://s3.waw3-1.cloudferro.com/mdl-native-10/native/ANTARCTIC_OMI_SI_extent/antarctic_omi_si_extent_202207/antarctic_omi_si_extent_19930115_P20220328.nc"
                    },
                    {
                      "service_name": "omi-arco",
                      "uri": "https://s3.waw3-1.cloudferro.com/mdl-arco-time-001/arco/ANTARCTIC_OMI_SI_extent/antarctic_omi_si_extent_202207/omi.zarr"
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }

You can also use the 'all' shortcut to return all the fields and then exclude if necessary:

.. code-block:: bash

    copernicusmarine describe -r all --exclude-fields uri,product_id,dataset_id,service_name,descrpition,keywords

.. note::

    By default all fields are now shown in the output. In v1, "keywords", "description" and "datasets" were not shown by default.
    To have a similar output as before, you can exclude them with the option ``--exclude-fields datasets,description,keywords``.

> 

Author: I was wondering if we should put a little warning saying something like: notice that field `datasets` is different than the field `dataset_id`. One is the parent of many other fields!

Reviewer: Maybe in the usage of the describe but not in the migration page

    copernicusmarine get --overwrite

Option ``--show-outputnames`` deleted
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
Before, this option would allow to output the name of the files.
We included such names in the :class:`copernicusmarine.ResponseGet` object that results for the call (either dry-run or not).

.. code-block:: bash

    # In v1
    copernicusmarine get -i cmems_mod_arc_bgc_my_ecosmo_P1D-m --filter "*202105/2021/12*" --show-outputnames > output.json

    # In v2
    copernicusmarine get -i cmems_mod_arc_bgc_my_ecosmo_P1D-m --filter "*202105/2021/12*" -r file_path > output.json

In the Python interface, the ``file_path`` key of the response object contains the same information that was in the ``--show-outputnames`` option.

.. code-block:: python

    # In v2
    response_get = copernicusmarine.get(...)
    files_local_paths = [file_get.file_path for file_get in response_get.files]

Subset
------------------

For more information, please see :ref:`the documentation page of the subset function <subset-page>`.

:class:`copernicusmarine.ResponseSubset` as object of the response
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

> Output of the get function has been changed. It is now a :class:`copernicusmarine.ResponseSubset` object in the Python interface or as a

Author: ```suggestion
Output of the subset function has been changed. It is now a :class:`copernicusmarine.ResponseSubset` object in the Python interface or as a
```

> json object in the command line interface. It used to be a list of paths to the downloaded files.

Author: ```suggestion
json object in the command line interface. It used to be a path to the downloaded file.
```
For the subset there was always only one file!

.. code-block:: python

    # In v1

>     subset_file_paths = copernicusmarine.get(...)
>     print(type(subset_file_paths))  # <class 'list'>
> 

Author: ```suggestion
    subset_path = copernicusmarine.get(...)
    print(type(subset_path))  # <class 'Path'>

```
Not actually sure what is the class of this before.

    # In v2

>     response_get = copernicusmarine.get(...)

Author: ```suggestion
    response_subset = copernicusmarine.subset(...)
```

>     print(type(response_get))  # <class 'copernicusmarine.ResponseSubset'>

Author: ```suggestion
    print(type(response_subset))  # <class 'copernicusmarine.ResponseSubset'>
```

    from copernicusmarine import ResponseSubset  # Can be imported like this for typing

    def my_function(response: ResponseSubset):
        pass

In the command line interface, it is possible to filter the result using the ``--response-fields`` option.
The input of the option is a comma-separated list of the fields to be included in the output.
The available fields are the name of the fields of the :class:`copernicusmarine.ResponseSubset` object.

.. code-block:: bash

>     copernicusmarine get ... -r minimum,data_transfer_size > request-metadata.json

Author: ```suggestion
    copernicusmarine subset ... -r minimum,data_transfer_size > request-metadata.json
```

Option ``--vertical-dimension-output`` renamed
""""""""""""""""""""""""""""""""""""""""""""""""

The option ``--vertical-dimension-output`` has been renamed to ``--vertical-axis``.
It now takes a string as input, either ``depth`` or ``elevation``.

.. code-block:: bash

    # To get elevation instead of depth (default is depth)
    # In v1
    copernicusmarine subset --vertical-dimension-output True

    # In v2
    copernicusmarine subset --vertical-axis elevation

Option ``--overwrite-output-data`` renamed to ``--overwrite``
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

The option ``--overwrite-output-data`` has been deleted, use directly ``--overwrite`` instead.

.. code-block:: bash

    # In v1

>     copernicusmarine get --overwrite-output-data
> 
>     # In v2
>     copernicusmarine get --overwrite

Author: ```suggestion
    copernicusmarine subset --overwrite-output-data

    # In v2
    copernicusmarine subset --overwrite
```

    copernicusmarine subset --netcdf-compression-level 5

Login
------------------

For more information, please see :ref:`the documentation page of the login function <login-page>`.

Options ``--overwrite`` and ``--overwrite-configuration-file`` renamed to ``--force-overwrite``
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

The options ``--overwrite`` and ``--overwrite-configuration-file`` have been renamed to ``--force-overwrite``. The option is still the same.

.. code-block:: bash

    # In v1
    copernicusmarine login --overwrite
    # or
    copernicusmarine login --overwrite-configuration-file

    # In v2
    copernicusmarine login --force-overwrite

Option ``--skip-if-user-logged-in`` deleted
"""""""""""""""""""""""""""""""""""""""""""""

The option ``--skip-if-user-logged-in`` has been deleted.
The option ``--check-credentials-valid`` can be used to check that the credentials are correctly set.

Now, we can do the login like this in our notebooks:

.. code-block:: python

    # In v1
    copernicusmarine.login(skip_if_user_logged_in=True)

    # In v2
    if not copernicusmarine.login(check_credentials_valid=True):

>         copernicusmarine.login()

Author: should we also add now the last function you created? for the path of the credentials? now there are 2 of those, no?

Reviewer: Yeah but shouldn't a breaking change because it's new
